<?xml version="1.0"?>
<launch>

	<rosparam command="load" file="$(find planning_control)/config/waypoints.yaml" />
	<rosparam command="load" file="$(find planning_control)/config/objects.yaml" />

	<!-- ROSPlan -->
	<include file="$(find rosplan_planning_system)/launch/interfaced_planning_system.launch" >
		<arg name="domain_path"		value="$(find planning_control)/pddl_files/domain.pddl" />
		<arg name="problem_path"	value="$(find planning_control)/pddl_files/problem.pddl" />
	</include>

	
	<!-- move_base action interface -->
	<node name="rosplan_interface_move_base" pkg="planning_control" type="moveBaseInterface" respawn="false" output="screen">
		<param name="knowledge_base"		value="rosplan_knowledge_base" />
		<param name="pddl_action_name"		value="move" />
		<param name="action_dispatch_topic" value="/rosplan_plan_dispatcher/action_dispatch" />
		<param name="action_feedback_topic" value="/rosplan_plan_dispatcher/action_feedback" />
	</node>

	<node name="rosplan_interface_pick" pkg="planning_control" type="pickInterface" respawn="false" output="screen">
		<param name="knowledge_base"		value="rosplan_knowledge_base" />
		<param name="pddl_action_name"		value="pick" />
		<param name="action_dispatch_topic" value="/rosplan_plan_dispatcher/action_dispatch" />
		<param name="action_feedback_topic" value="/rosplan_plan_dispatcher/action_feedback" />
	</node>

	<node name="rosplan_interface_place" pkg="planning_control" type="placeInterface" respawn="false" output="screen">
		<param name="knowledge_base"		value="rosplan_knowledge_base" />
		<param name="pddl_action_name"		value="place" />
		<param name="action_dispatch_topic" value="/rosplan_plan_dispatcher/action_dispatch" />
		<param name="action_feedback_topic" value="/rosplan_plan_dispatcher/action_feedback" />
	</node>

	<!-- tiago_example_skills -->
	<include file="$(find tiago_example_skills)/launch/server.launch" >
	</include>

</launch>
